src/webui/gitutils.jl

```diff
@@ -40,130 +66,180 @@ is_success(res::AuthFailure) = false
 # The criteria is simply whether the user is a collaborator for user-owned repos,
 # or whether they're an organization member or collaborator for organization-owned repos.
+# the fetch argument is only used in testing
 isauthorized(u, repo) = AuthFailure("Unkown user type or repo type")
-function isauthorized(u::User{GitHub.User}, repo::GitHub.Repo)
-    if !get(CONFIG, "allow_private", false)
-        repo.private && return AuthFailure("Repo $(repo.name) is private")
-    end
-
-    if repo.private
-        forge = PROVIDERS["github"].client
-    else
-        forge = u.forge
-    end
-
-    if repo.organization === nothing
-        hasauth = @gf @mock is_collaborator(forge, repo.owner.login, repo.name, u.user.login)
-        if something(hasauth, false)
-            return AuthSuccess()
-        else
-            return AuthFailure("User $(u.user.login) is not a collaborator on repo $(repo.name)")
-        end
-    else
-        # First check for organization membership, and fall back to collaborator status.
-        ismember = @gf @mock is_member(forge, repo.organization.login, u.user.login)
-        hasauth = something(ismember, false) ||
-            @gf @mock is_collaborator(forge, repo.organization.login, repo.name, u.user.login)
-        if something(hasauth, false)
-            return AuthSuccess()
-        else
-            return AuthFailure("User $(u.user.login) is not a member of the org $(repo.organization.login) and not a collaborator on repo $(repo.name)")
-        end
+function isauthorized(u::User{GitHub.User}, repo::GitHub.Repo, fetch = true)
+    !get(CONFIG, "allow_private", false) && repo.private &&
+        return AuthFailure("Repo $(repo.name) is private")
+    jforge = provider(repo).client
+    if fetch
+        repo = @gf_q @mock get_repo(u.forge, repo.owner.login, repo.name)
     end
+    # Users with push access can always release their package
+    repo !== nothing && repo.permissions.push && return AuthSuccess()
+    # Collaborators can always release their package
+    # checking collaborators requires push access for both the connection and the acct
+    # check with each connection in case user's connection does not have push permission
+    ((@gf_bool @mock is_collaborator(u.forge, repo.owner.login, repo.name, u.user.login)) ||
+        (@gf_bool @mock is_collaborator(jforge, repo.owner.login, repo.name, u.user.login))) &&
+        return AuthSuccess()
+    # If the repo is not in an org, the user does not have sufficient access to create a PR
+    repo.organization === nothing &&
+        return AuthFailure("User $(u.user.login) is not a collaborator on repo $(repo.name)")
+    # Otherwise the user must be a member of the organization
+    # verify with user's connection, falling back to JuliaHub's connection
+    (something((@gf_q @mock is_member(u.forge, repo.organization.login, u.user.login)), false) ||
+        something((@gf_q @mock is_member(jforge, repo.organization.login, u.user.login)), false)) &&
```

Comment:
These two lines can be simplified. `something((@gf_q f()), false)` is same as `@gf_bool f()`.

Reply:
pushed that change